pkg/configurator/methods.go

```diff
@@ -162,6 +165,22 @@ func (c *Client) GetServiceCertValidityPeriod() time.Duration {
 	return validityDuration
 }
 
+// GetIngressGatewayCertValidityPeriod returns the validity duration for ingress gateway certificates, and a default in case of unspecified or invalid duration
+func (c *Client) GetIngressGatewayCertValidityPeriod() time.Duration {
```

Comment:
What's the use case to use a default if unspecified? I'd rather make this field required in the CRD spec vs assuming user intent.

Reply:
If GetIngressGatewaCertValidityPeriod() is somehow called when the ingressGateway field isn't specified should it return an error? Currently, If the ingressGateway ValidityDuration field is updated and the duration is invalid the ingressGatewayCert won't be updated. On rotation of the ingressGateway cert we'll attempt to get the duration from the meshConfig, but it will be invalid. Do you think it makes sense to return a default duration in this case?

Reply:
If an ingressGateway is specified in the MeshConfig then the duration is required. We could specify a default instead of making it required? I guess ideally we would validate the MeshConfig and prevent invalid durations before they are applied.

Reply:
If it's a required property, why do we need a default? Is there a default to simplify the code path?

Reply:
If the ingressGateway field is removed/unset, the behavior as implemented in pkg/ingress/gateway.go is to call ReleaseCertificate(). So rotation should not happen for such a scenario?

Reply:
> If it's a required property, why do we need a default? Is there a default to simplify the code path?

I am not sure what you mean. Are you asking if I am using a default value to make things simpler?

Reply:
Yes. Additionally,

> On rotation of the ingressGateway cert we'll attempt to get the duration from the meshConfig, but it will be invalid.

I am wondering if we ever want to use a default during rotation for the ingress cert. I don't see a need to rotate the ingress gateway cert when the state of the gateway cert spec is deemed invalid. Would it be easier to prevent a misconfig? 

This is not a blocking comment. I can understand if there are a few corner cases being guarded against through defensive programming here.

Reply:
Also I would imagine, if the updated value is invalid, we should continue to use the older value encoded in the previous cert and not a default?

Reply:
If the ingress gateway cert's validity duration is set in the meshConfig to an invalid duration the cert will not be reissued but the invalid duration will remain set in the meshConfig. When the ingress gateway cert expires should the previous validity duration be used? Prior to my change, the service cert validity duration would have been used for the ingress gateway cert on expiration. I could add a validity duration field to the `Certificate` struct or I could add validation for the MeshConfig, but I think that is beyond the scope of this PR.

Reply:
Could this be addressed in a follow-up PR? I think this could be addressed when we add validation for the MeshConfig. I am also happy to make the changes to include the validity duration as a field in `Certificate` or make those changes in a follow-up PR as well.

Reply:
Yes, sure. This comment is not blocking in nature.